### PR TITLE
MM-37531: Make subheaders in RHS sticky

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Subheaders are sticky now

https://user-images.githubusercontent.com/1490756/142899480-93627b3e-1075-4c62-b736-203862f247ed.mp4

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-37531

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Comment by calebroseland** [comment]
Oooh 😃

File: webapp/src/components/collapsible_checklist.tsx [hunk 1]
```diff
@@ -116,7 +116,17 @@ const ProgressLine = styled.div<{ width: number }>`
...
+    background-color: var(--center-channel-bg);
+    z-index: 1;
+    position: sticky;
+    top: 48px; // height of rhs_checklists MainTitle
+`;
+
 const Horizontal = styled.div`
+    background-color: var(--center-channel-color-04);
+    border-radius: 4px 4px 0 0;
+    border: 1px solid var(--center-channel-color-08);
```
Comment (agarciamontoro) [inline webapp/src/components/collapsible_checklist.tsx:129]: Should we try to always use the rgba ones?
```suggestion
    background-color: rgba(var(--center-channel-color-rgb, 0.04));
    border-radius: 4px 4px 0 0;
    border: 1px solid rgba(var(--center-channel-color-rgb, 0.08));
```

File: webapp/src/components/rhs/rhs_checklists.tsx [hunk 2]
```diff
@@ -233,23 +237,38 @@ const InnerContainer = styled.div`
...
+    background-color: var(--center-channel-bg);
+    z-index: 2;
+    position: sticky;
+    top: 0;
+`;
+
 const MainTitle = styled.div`
-    ${SemiBoldHeading}
+    ${SemiBoldHeading} {
+    }
```
Comment (agarciamontoro) [inline webapp/src/components/rhs/rhs_checklists.tsx:256]: Why do we need the curly braces?

File: webapp/src/components/rhs/rhs_checklists.tsx [hunk 3]
```diff
@@ -233,23 +237,38 @@ const InnerContainer = styled.div`
...
     font-size: 16px;
     line-height: 24px;
-    margin: 0 0 0 8px;
+    padding: 12px 0 12px 8px;
 `;
 
 const ChecklistContainer = styled.div`
     background-color: var(--center-channel-bg);
+    border-radius: 0 0 4px 4px;
+    border:  1px solid var(--center-channel-color-08);
```
Comment (agarciamontoro) [inline webapp/src/components/rhs/rhs_checklists.tsx:270]: Same here
```suggestion
    border:  1px solid rgba(var(--center-channel-color-rgb, 0.08));
```

**Review by agarciamontoro — APPROVED** [review]
Soooo smooth! I love it, thanks :)

I left a couple of non-blocking comments about using explicit rgba instead of the variables with opacity.

File: webapp/src/components/rhs/rhs_checklists.tsx (hunk 2 above)
Comment (cpoile) [reply webapp/src/components/rhs/rhs_checklists.tsx:256]: This is the correct eslint-approved way of doing this (it was always showing up in red in my editor, now it doesn't!)

Comment (agarciamontoro) [reply webapp/src/components/rhs/rhs_checklists.tsx:256]: Oh :eyes: If eslint approves it, then it must be good.

File: webapp/src/components/collapsible_checklist.tsx [hunk 4]
```diff
@@ -116,7 +116,17 @@ const ProgressLine = styled.div<{ width: number }>`
...
 
+const HorizontalBG = styled.div`
+    background-color: var(--center-channel-bg);
+    z-index: 1;
+    position: sticky;
+    top: 48px; // height of rhs_checklists MainTitle
+`;
+
 const Horizontal = styled.div`
+    background-color: rgba(var(--center-channel-color-rgb, 0.04));
```
Comment (calebroseland) [inline webapp/src/components/collapsible_checklist.tsx:127]:
```suggestion
    background-color: rgba(var(--center-channel-color-rgb), 0.04);
```

File: webapp/src/components/collapsible_checklist.tsx [hunk 5]
```diff
@@ -116,7 +116,17 @@ const ProgressLine = styled.div<{ width: number }>`
...
+    border-radius: 4px 4px 0 0;
+    border: 1px solid rgba(var(--center-channel-color-rgb, 0.08));
```
Comment (calebroseland) [inline webapp/src/components/collapsible_checklist.tsx:129]:
```suggestion
    border: 1px solid rgba(var(--center-channel-color-rgb), 0.08);
```

File: webapp/src/components/rhs/rhs_checklists.tsx [hunk 6]
```diff
@@ -233,23 +237,38 @@ const InnerContainer = styled.div`
...
     font-size: 16px;
     line-height: 24px;
-    margin: 0 0 0 8px;
+    padding: 12px 0 12px 8px;
 `;
 
 const ChecklistContainer = styled.div`
     background-color: var(--center-channel-bg);
+    border-radius: 0 0 4px 4px;
+    border:  1px solid rgba(var(--center-channel-color-rgb, 0.08));
```
Comment (calebroseland) [inline webapp/src/components/rhs/rhs_checklists.tsx:270]:
```suggestion
    border:  1px solid rgba(var(--center-channel-color-rgb), 0.08);
```

**Review by calebroseland — CHANGES_REQUESTED** [review]
LGTM! Just a quick fix for the CSS/parenthesis to fix the theming

**Review by calebroseland — APPROVED** [review]

**Pull request merged**